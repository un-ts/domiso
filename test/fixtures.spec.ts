import fs from 'node:fs/promises'
import path from 'node:path'
import { fileURLToPath } from 'node:url'

import { sanitize, sanitizeSvg } from 'domiso'

const fixtures = path.resolve(fileURLToPath(import.meta.url), '../fixtures')

describe('fixtures', () => {
  test('html', async () => {
    const htmlFixtures = path.resolve(fixtures, 'html')
    const files = await fs.readdir(htmlFixtures)
    for (const file of files) {
      expect(
        sanitize(await fs.readFile(path.resolve(htmlFixtures, file), 'utf8')),
      ).toMatchSnapshot()
    }
  })

  test('fragment', async () => {
    const fragmentFixtures = path.resolve(fixtures, 'fragment')
    const files = await fs.readdir(fragmentFixtures)
    for (const file of files) {
      expect(
        sanitize(
          await fs.readFile(path.resolve(fragmentFixtures, file), 'utf8'),
          true,
        ),
      ).toMatchSnapshot()
    }
  })

  test('svg', async () => {
    const svgFixtures = path.resolve(fixtures, 'svg')
    const files = await fs.readdir(svgFixtures)
    for (const file of files) {
      expect(
        sanitizeSvg(await fs.readFile(path.resolve(svgFixtures, file), 'utf8')),
      ).toMatchSnapshot()
    }
  })
})
