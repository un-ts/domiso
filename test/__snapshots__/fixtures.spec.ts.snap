// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > fragment > is.html 1`] = `
"<b is="">bar</b>
"
`;

exports[`fixtures > fragment > style.html 1`] = `
"
<style>
  .x {
    border: 1px solid #000;
  }
</style>

"
`;

exports[`fixtures > fragment > svg-g.html 1`] = `
"<div id="11">
  <svg xmlns="http://www.w3.org/2000/svg">
    </svg>//["'\\\`--&gt;]]&gt;]
</div>
"
`;

exports[`fixtures > fragment > template.html 1`] = `
"<select>
  <template>
    <img src="x">
  </template>
</select>
"
`;

exports[`fixtures > html > basic.html 1`] = `
"<html lang="en"><head>
    
    
    
    <title>Document</title>
  </head>
  <body>
    <img src="/whatever-error">
    
  

</body></html>"
`;

exports[`fixtures > svg > basic.svg 1`] = `
"<svg>
  <text>Hello World!</text>
  
</svg>"
`;

exports[`fixtures > svg > image-href.svg 1`] = `
"<svg width="24" height="16" viewBox="0 0 24 16" xmlns="http://www.w3.org/2000/svg">
  
    
    
      11111
      
      
    
  
</svg>"
`;

exports[`fixtures > svg > xml.svg 1`] = `
"<svg class="icon" viewBox="0 0 1024 1024" xmlns="http://www.w3.org/2000/svg" width="200" height="200">
  
  
</svg>"
`;
